core/google/api/core/retry.py

```diff
@@ -226,11 +231,14 @@ def __init__(
         self._maximum = maximum
         self._deadline = deadline
 
-    def __call__(self, func):
+    def __call__(self, func, on_error=None):
```

Comment:
Does it make sense to pass it in here or in the constructor? I can see the argument for it being here because it (so far) only makes sense for the function that actually uses the retry to specify this.

Reply:
I tried it that way first, but then realized one might instantiate the `Retry` instance globally, and then use it in places where one wanted to use local-scope objects.

Reply:
Cool - makes sense.